README.md

```diff
@@ -28,41 +35,48 @@ Get started writing your application by modifying these key files:
 
 Associated build configuration files and tests are discussed along with the [project structure](#project-structure) below.
 
-### Build
-
-Running the following will build the [zkVM guest program].
+## Test Your Project
+With this Foundry template, you can write a zkVM Rust program in the `/methods/guest` directory which Solidity contracts in  `/contracts` can call into and in return receive a proof or 'receipt' of execution.
+### Test your zkVM program
+To check if your zkVM program will compile, and generate any errors if not, run the Rust compiler with
 
 ```bash
 cargo build
 ```
 
-While the following will build your Ethereum contracts.
-
+If you've written tests in your zkVM progam, run them with
 ```bash
-forge build
+cargo test
 ```
+***Hint:*** *To learn more about our RISC-V zkVM [visit the docs](https://dev.risczero.com/zkvm) or for a thorough walkthrough, follow the [Factors Tutorial here](https://github.com/risc0/risc0/tree/main/examples/factors#tutorial).*
 
-### Test
-
-Running the following will run the Ethereum contract tests using your [zkVM guest program], but without running
-the expensive computations required to prove its behavior in zero-knowledge.
+### Test your project end to end
+To test your project, end to end, including both your Solidity contracts and their interaction with your zkVM program, run
 
 ```bash
 forge test
 ```
 
-For testing with [proof] generation, which might take some time to complete, execute the following command instead:
-```bash
-BONSAI_PROVING=local forge test
-```
+***Hint:*** *To learn more about Foundry's `forge` command and the other helpful utilities Foundry provides, visit their docs: https://book.getfoundry.sh/forge/tests.*
+
+## Proving Modes
+The foundry template supports three different proving modes:
+1. `none` - By default, only the [executor](https://www.dev.risczero.com/docs/terminology#executor) runs your zkVM program and no proof is generated. Because there is no proving, this will be the fastest way to test. 
+2. `local` - A proof of execution generated on your local machine. 
+3. `bonsai` - A proof of execution is generated by the Bonsai API. 
 
-For offloading your [proof] requests to a local Bonsai instance, you can execute the tests as follows:
+Configure your preferred mode by setting the `BONSAI_PROVING` env variable.
+```bash
+export BONSAI_PROVING="local"
+``` 
```

Comment:
As a software noob this was a bit jarring for me. What do I do once I set the mode? 
```suggestion
``` 
After setting your preferred proving mode, use `cargo build`, `cargo test`, and `forge test` as explained above.
```